src/VisualStudio/Core/Def/Implementation/DebuggerIntelliSense/DebuggerTextView.cs

```diff
@@ -364,6 +364,25 @@ public void DisconnectFromIntellisenseControllers()
             }
         }
 
+        public void QueuePostLayoutAction(Action action) => _innerTextView.QueuePostLayoutAction(action);
+
+        public bool TryGetTextViewLines(out ITextViewLineCollection textViewLines) => _innerTextView.TryGetTextViewLines(out textViewLines);
+
+        public bool TryGetTextViewLineContainingBufferPosition(SnapshotPoint bufferPosition, out ITextViewLine textViewLine)
+        {
+            // This implementation confirms with the Editor implementation.
+            try
+            {
+                textViewLine = this.GetTextViewLineContainingBufferPosition(bufferPosition);
+                return textViewLine != null;
+            }
+            catch
```

Comment:
this catch-all really worries me.  why is this necessary?

Reply:
It's a paranoid check to make sure we don't have any throws from a TryGet... method. We've had a few cases show up where getting a textViewLine can cause an InvalidSnapshotException, and while we've fixed that, it seemed unwise to just have an unguarded call to a method that has had known exceptions in the past. In the editor side, we actually have a telemetry firing path in the catch so we can track down bad issues, but here in the shim it didn't make sense to spin up all the telemetry stuff, so we just added the guards to keep things from throwing.

Reply:
This is an exact copy of the Editor implementation.
We're waiting for them to provide a fix for implementation of _innerTextView.TryGetTextViewLineContainingBufferPosition() but now there is a breaking change in the interface on their side. We decided to follow their approach exactly as it is.

---
In reply to: [223945696](https://github.com/dotnet/roslyn/pull/30419#discussion_r223945696) [](ancestors = 223945696)